src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -207,6 +258,45 @@ def _get_requirements_and_protocol(self,
         else:
             raise NotImplementedError(str(backend) + backend_msg)
 
+    def _get_run_config(self, backend):
+        if backend in CPUS:
+            pdict = {self.solver_process: self.solver_model,
+                     ReadGate: ReadGatePyModel,
+                     Dense: PyDenseModelFloat,
+                     StochasticIntegrateAndFire: StochasticIntegrateAndFireModel
+                     }
+            run_cfg = Loihi1SimCfg(exception_proc_model_map=pdict,
+                                   select_sub_proc_model=True)
+        elif backend in NEUROCORES:
+            pdict = {self.solver_process: self.solver_model,
+                     ReadGate: ReadGateCModel,
+                     Dense: NcModelDense,
+                     StochasticIntegrateAndFire:
+                         StochasticIntegrateAndFireModelSCIF,
+                     QuboScif: NcModelQuboScif,
+                     }
+            pre_run_fxs, post_run_fxs = [], []
+            run_cfg = Loihi2HwCfg(exception_proc_model_map=pdict,
+                                  select_sub_proc_model=True,
+                                  pre_run_fxs=pre_run_fxs,
+                                  post_run_fxs=post_run_fxs
+                                  )
+        else:
+            raise NotImplementedError(str(backend) + backend_msg)
+        return run_cfg
+
+    def _validated_cost(self, target_cost):
+        if target_cost % int(target_cost) != 0:

```

Comment:
This throws an error if the if the target_cost is 0 - which is actually the default value! Thus, I would change to
`if target_cost !=0 and target_cost % int(target_cost) != 0:`